src/lib/viz/dataview/mode/DataViewLocal.ts

```diff
@@ -60,9 +55,8 @@ export class DataViewLocal extends DataViewMode {
     }
   }
 
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars, class-methods-use-this
-  public setSpatialFilter(_spatialFilter: SpatialFilters) {
-    return undefined;
+  public setSpatialFilter(spatialFilter: SpatialFilters) {
+    this.useViewport = spatialFilter === 'viewport';
```

Comment:
I'm ok with this, but I guess that in the future we might need the whole spatialFilter stored